### PR TITLE
feat: add support for Chat Webhooks messages

### PR DESCRIPTION
**Pull request by maelgangloff**
## Describe your changes

Webhooks allow sending messages in many instant messaging software (Mattermost, Rocket.Chat, ...).
In addition to email alerts, it may be useful to add support for these messages by Webhook.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist before requesting a review

- [x] Commit names follow the [Conventional Commits](https://www.conventionalcommits.org/fr/v1.0.0/) convention
- [x] I have checked the entire code before submitting it
- [ ] I have updated the documentation related to my commits
- [x] My code does not generate errors

### CODE REVIEW

File: config/packages/notifier.yaml [hunk 1]
```diff
@@ -1,6 +1,14 @@
 framework:
     notifier:
         chatter_transports:
+            zulip: '%env(ZULIP_DSN)%'
```
Comment (vinceh121) [inline config/packages/notifier.yaml:4]: Do you need this if transports are dynamic?

File: src/MessageHandler/ProcessDomainTriggerHandler.php [hunk 2]
```diff
@@ -91,67 +102,29 @@ public function __invoke(ProcessDomainTrigger $message): void
...
+                $recipient = new Recipient($watchList->getUser()->getEmail());
+                $notification = new DomainUpdateNotification($this->sender, $event);
+
                 if (TriggerAction::SendEmail == $watchListTrigger->getAction()) {
-                    $this->sendEmailDomainUpdated($event, $watchList->getUser());
+                    $this->mailer->send($notification->asEmailMessage($recipient)->getMessage());
+                } elseif (TriggerAction::SendChat == $watchListTrigger->getAction()) {
+                    if (null !== $watchList->getWebhookDsn()) {
+                        foreach ($watchList->getWebhookDsn() as $dsnString) {
+                            $dsn = new \Symfony\Component\Notifier\Transport\Dsn($dsnString);
```
Comment (vinceh121) [inline src/MessageHandler/ProcessDomainTriggerHandler.php:114]: use use

File: assets/pages/tracking/WatchlistPage.tsx [hunk 3]
```diff
@@ -32,15 +33,23 @@ export default function WatchlistPage() {
...
         domains: string[],
-        emailTriggers: string[]
-        connector?: string
+        triggers: string[]
+        connector?: string,
+        dsn?: string[]
     }) => {
         const domainsURI = values.domains.map(d => '/api/domains/' + d.toLowerCase())
+        let triggers = values.triggers.map(t => ({event: t, action: 'email'}))
+        if (values.dsn !== undefined) triggers = [...triggers, ...values.triggers.map(t => ({
```
Comment (vinceh121) [inline assets/pages/tracking/WatchlistPage.tsx:42]: Space before and after blocks

```suggestion
        if (values.dsn !== undefined) {
                triggers = [...triggers, ...values.triggers.map(t => ({
```

File: assets/pages/tracking/WatchlistPage.tsx [hunk 4]
```diff
@@ -54,17 +63,24 @@ export default function WatchlistPage() {
...
```
Comment (vinceh121) [inline assets/pages/tracking/WatchlistPage.tsx:72]: Space before and after blocks

```suggestion
        if (values.dsn !== undefined) {
                triggers = [...triggers, ...values.triggers.map(t => ({
```

**Review by vinceh121 — CHANGES_REQUESTED** [review]

File: config/packages/notifier.yaml (hunk 1 above)
Comment (maelgangloff) [reply config/packages/notifier.yaml:4]: These settings won't create any problems I think. If there is ever sending from an internal DSN I prefer to keep it. These environment variables are not defined.

**Review by vinceh121 — APPROVED** [review]

**Comment by maelgangloff** [comment]
Thanks for your review @vinceh121 :tada:

**Pull request merged**